libyul/backends/evm/NoOutputAssembly.cpp

```diff
@@ -212,13 +212,13 @@ NoOutputEVMDialect::NoOutputEVMDialect(EVMDialect const& _copyFrom):
 	// them in one go, later reference pointers to this static vector
 	static std::vector<BuiltinFunctionForEVM> noOutputBuiltins = defineNoOutputBuiltins();
 
-	m_functions.reserve(m_functions.size());
```

Comment:
Hm, so this should [in theory](https://eel.is/c++draft/vector.capacity#4) never do anything, as `reserve` will only perform a reallocation if and only if `capacity()` is smaller than the new requested capacity. `size()` should thus technically always be smaller or equal to `capacity()`, as auto reallocation (i.e. capacity increase) is only performed when you want to insert an element that would exceed the container's capacity.

What I'm trying to say is that ASAN was likely wrong here, but this then begs another question - the old implementation appends builtin handles to `m_functions`, whereas the new one overwrites the current ones (which is good and correct) - but then, how come this wasn't caught in any tests? Do we even have tests for no output builtins?

Reply:
The issue was with `emplace_back` down below. This is indeed a no-op so I removed it as well. We are looping over `m_functions` and were simultaneously appending to it.

Reply:
Yeah, that part I understand - we had bugs like these before, which are usually quite insidious and can't be reproduced via tests quite well; my question was more so aimed at the semantic difference between the two (assume iterator invalidation is not an issue) - old implementation appends, where as the new one overwrites - this should in theory mean that in the old case `m_functions` will always have more elements than in the new (overwritten) version?

In any case, it's used in the compatibility checker, so no big deal, but still weird that it wasn't caught by *some* test. Although from what I can see, we have no such tests, so this makes perfect sense :)